cli/cmd/dashboard.go

```diff
@@ -55,6 +74,26 @@ var dashboardCmd = &cobra.Command{
 	},
 }
 
+func isDashboardAvailable(kubeAPI k8s.KubernetesApi) (bool, error) {
+	client, err := kubeAPI.NewClient()
+	if err != nil {
+		return false, err
+	}
+	fullUrl, err := kubeAPI.UrlFor(controlPlaneNamespace, "/services/web")
+	if err != nil {
+		return false, err
+	}
+	req, err := client.Get(fullUrl.String())
+	if err != nil {
+		return false, err
+	}
+	if req.StatusCode == http.StatusOK {
```

Comment:
Checkin on `200` creates some coupling between this check and how we happen to expose the dashboard. Imagine, for example, that a new release of Kubernetes changes URL formats but, to keep backwards compatibility, it automatically uses [one of the many redirect codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#3xx_Redirection) so that clients can automatically fetch the new URL. Do we know how the `http` package behaves inc ase of redirects? Maybe we should, instead, just make sure that the status code isnt `4xx` or `5xx`?

Reply:
That is a good point about the redirects, I am not sure if they are considered errors. Would it be safe to classify them as errors as well? Given that a `3xx` requests indicate
> the client must take additional action to complete the request

I would think we would want to stop attempting to open the dashboard considering that we may not know what additional work we have to do to open the dashboard. I will modify the test to check for `4xx` and `5xx` responses.

Reply:
I think that we should see what a unit test of this function tells us about 3xx and other conditions, then we can make an informed decision :)